### DISCUSSION TITLE
"POST /items/ HTTP/1.1" 422 Unprocessable Entity

### DISCUSSION THREAD

**Discussion by pawanm09** [fastapi/fastapi#7375] [Questions]
Testing the post request of Fastapi. 

But, every time I get 422 Unprocessable Entity Error. 

```
from typing import Optional

from fastapi import FastAPI
from pydantic import BaseModel
import uvicorn


class Item(BaseModel):
    name: str
    description: Optional[str] = None
    price: float
    tax: Optional[float] = None


app = FastAPI()


@app.post("/items/")
async def create_item(item: Item):
    print(item)
    return item

```
Running it on gcp using : `uvicorn testing:app --host 0.0.0.0 --port 5000 `

**Request Structure:**
Post Request on the route:` http://35.132.89.173:5000/items/`
body: { "name": "nmsaey", "price":34}
header: { Content-Type: application/json} 

**Version**:
python: 3.6.9
fastapi: 0.60.1
pydantic: 1.6.1

**Comment by marodev** [comment]
The payload `{ "name": "nmsaey", price:34}` doesn't seem to be properly json formatted.

Try to put price in quotation marks: `{"name": "nmsaey", "price": 34}`

**Comment by pawanm09** [comment]
@marodev  Sorry, it was just a typo.

price is in quotes. I will update the question.

**Comment by marodev** [comment]
How do you send the HTTP POST request? With a tool like Postman or by using a Python script?

**Comment by pawanm09** [comment]
@marodev 
Through Postman

**Comment by Kludex** [comment]
It works just fine here, command used to test:
```
curl -X POST "http://0.0.0.0:8000/items/" -H  "accept: application/json" -H  "Content-Type: application/json" -d "{\"name\":\"nmsaey\",\"price\":34}"
```

**Comment by ArcLightSlavik** [comment]
>Running it on gcp

I'll assume you are running it inside a virtual machine since the host looks very similar to what a vm makes `http://35.132.89.173:5000/items/` 

Make sure when creating the vm you click the 'Allow HTTP traffic' option
If that doesn't work check if you still have the vm ssh terminal open # yes dumb but from personal experience (:man_facepalming:) you should try

**Comment by marodev** [comment]
Works for me as well. Does it look similar to this:

<img width="633" alt="image" src="https://user-images.githubusercontent.com/23633010/88365121-57774480-cd85-11ea-86e4-c7b4153eed6a.png">


<img width="623" alt="image" src="https://user-images.githubusercontent.com/23633010/88365088-3adb0c80-cd85-11ea-88d5-98d9c73863ff.png">

**Comment by pawanm09** [comment]
Thanks for your response 

The problem was in Postman.

Everything was fine using code.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :clap: :bow: 

Thanks for reporting back and closing the issue :+1:

**Comment by eskayamadeus** [comment]
The Postman solution worked for me as well.

**Comment by ThiruSudha** [comment]
firast = item.name
second   =  item.price

return{"name":first,"price":second}

try this working fine
